File: packages/python/examples/02_post.py
"""テキスト投稿の例。投稿系のエンドポイントは X-Jwt を必要とします。

generate_x_jwt は client のメソッドではなくトップレベル関数で、
client.api_version_key を渡します（呼び出しごとに再生成してください）。

    YAY_EMAIL=... YAY_PASSWORD=... python examples/02_post.py
"""

import asyncio
import os

import yaylib


async def main() -> None:
    async with yaylib.Client() as client:
        await client.login_with_email(
            email=os.environ["YAY_EMAIL"],
            password=os.environ["YAY_PASSWORD"],
        )

        post = await client.create_post(
            x_jwt=yaylib.generate_x_jwt(api_version_key=client.api_version_key),
            post_type="text",
            text="hello from yaylib",
        )
        print("posted:", post.id)


asyncio.run(main())

File: packages/typescript/examples/02_post.ts
// テキスト投稿の例。投稿系のエンドポイントは X-Jwt を必要とします
// （client.generateXJwt() で取得し、そのまま渡します）。
//
//   YAY_EMAIL=... YAY_PASSWORD=... npx tsx examples/02_post.ts
import { Client, generateXJwt } from "yaylib";

const client = new Client();

await client.loginWithEmail({
  email: process.env.YAY_EMAIL!,
  password: process.env.YAY_PASSWORD!,
});

const post = await client.createPost({
  xJwt: generateXJwt({ apiVersionKey: client.apiVersionKey }),
  postType: "text",
  text: "hello from yaylib",
});

console.log("posted:", post.id);
